src/charm.py

```diff
@@ -430,7 +430,8 @@ def _on_show_proxied_endpoints(self, event: ActionEvent):
             try:
                 result.update(provider.proxied_endpoints)
             except Exception as e:
-                msg = f"failed to fetch proxied endpoints from provider {provider} with error {e}."
+                name = provider.relations[0].app.name
```

Comment:
Could this line raise in some circumstances?

Reply:
On second consideration, it might. Added a getattr-guard.
Also I realized it's not necessarily the 0th relation that's broken. We need to put this message in each ingress lib separately or we won't know which one is broken.